atlas/readme.md

```diff
@@ -0,0 +1,325 @@
+
+That is, we will want to mark pieces of the graph in arbitrary ways. For example, an organization may want to mark certain fields as containing personally identifying information. Naturally, you can do this with a custom directive on the field:
+
+```graphql
+type User {
+  address: String @userdata(privacy: PII)
+}
+```
+
+But constellation also allows us to do this in another schema entirely:
+
+```graphql
+extend type users.User {
+  users.address: String @userdata(privacy: PII)
+}
+```
+
+Here we're using hypothetical starql syntax, but it should be possible for such post-schema directives to be applied by other tooling as well.
+
+## Structure
+
+### Graphs are made of Points
+
+We use the term *point* to refer to all the stuff in a graph. These are all points:
+- types
+- inputs
+- interfaces
+- fields
+- unions
+- enums
+- scalars
+- directives
+- schemas
+
+### `atlas:` urls uniquely address points
+
+Technically atlas URLs are URIs or maybe URNs but [WHATWG basically thinks the distinction is unnecessary and confusing](https://url.spec.whatwg.org/#goals) so we'll just follow their lead and call them URLs.
+
+They take this form:
+
+```
+atlas:specifier:path?layer
+```
+
+Every part except the initial `atlas:` is optional.
+
+Some examples:
+
+```
+# these are all in the main atlas
+
+## a schema
+atlas::spotify
+
+## a point within the schema
+atlas::spotify.User
+
+## a field
+### note: still thinking about the syntax here
+atlas::spotify.id/spotify.User=spotify.ID
+
+## a scalar
+atlas::genius.ID
+
+## an extension field
+atlas::lyrical.geniusId/spotify.Song=genius.ID
+
+# these are in different atlases
+## identified by aliases
+atlas:corp-internal:svc.secrets/User=String
+
+## identified by atlas id
+atlas:[3deff29adabade98f3234acb9320]:svc.secrets/User=String
+```
+
+#### Atlas URLs start off by specifying the atlas
+
+Atlases can be specified either by *id* or *alias*. IDs are specified in `[brackets]`, aliases are bare words:
+
+Specified by id:
+```
+atlas:[3deff29adabade98f3234acb9320]:
+atlas:[9eff9asdd9329d89d9cbd9cf3dde]:
+```
+
+Specified by alias:
+```
+atlas:main:
+atlas:acme-corp-internal:
+```
+
+Aliases are useful only within a context in which they're defined—for example, in a workspace with a [`.atlas` directory](#atlas-root).
+
+You can also omit the specifier:
+
+```
+atlas::
+```
+
+If omitted, the id of the public atlas will be used. These are equivalent:
+
+```
+atlas::
+atlas:[3deff29adabade98f3234acb9320]:
+# note, this is a lie 🖕🏽
+# we have not created a public atlas yet
+# therefore there is no public atlas id
+# but once there is, it will go here.
+```
+
+**Discussion Q:** Another option here is that `atlas::` is equivalent to `atlas:main:` or some other alias. 
+
+#### Dotted paths identify points
+
+```
+atlas::genius.User
+atlas::nyt.Article
+```
+
+#### Edges are located within their declaring schema
+
+All `id` fields declared on any type *by* the `genius` schema:
+
+```
+atlas::genius.id
```

Comment:
what does this do? Since `id` could have any arbitrary number of return types I'm not clear what this does?